runtime/io/io.cpp

```diff
@@ -0,0 +1,389 @@
+#include <gmp.h>
+#include <cstring>
+#include <stdexcept>
+
+#include "fcntl.h"
+#include "unistd.h"
+#include "runtime/alloc.h"
+#include "runtime/header.h"
+
+extern "C" {
+
+#define KCHAR char
+#define ERRTAG(err) "Lbl'hash'" #err
+
+  mpz_ptr move_int(mpz_t);
+
+  static block dotK = {getBlockHeaderForSymbol((((uint64_t)getTagForSymbolName("dotk")) << 32) | 1), {}};
```

Comment:
this should not be a block, this should be a block * and should also just cast the ored value to a pointer rrather than calling getBlockHeaderForSymbol.